chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -554,39 +554,58 @@ void BraveRenderViewContextMenu::ExecuteAIChatCommand(int command) {
       ai_chat::features::kAIChatSSE.Get() && IsRewriteCommand(command) &&
       !source_web_contents_->GetUserData(kAIChatRewriteDataKey);
 
-  if (!rewrite_in_place) {
+  auto [action_type, p3a_action] = GetActionTypeAndP3A(command);
+  auto selected_text = base::UTF16ToUTF8(params_.selection_text);
+
+  if (rewrite_in_place) {
+    source_web_contents_->SetUserData(kAIChatRewriteDataKey,
```

Comment:
seems like this would be better handled inside the tab helper so it gets cleared when the page navigates? I think we have way too much logic for aichat embedded in RenderViewContextMenu here that should be encapsulated within aichat

Reply:
also are there tests for this somewhere? I'm not seeing anything, but searching github PRs sucks so I could be missing something

Reply:
nvm, I found them

Reply:
Sure but this is not added in this PR cc @yrliou

Reply:
can we create follow-up issues to track this stuff?

Reply:
https://github.com/brave/brave-browser/issues/41171